From 9217aafa0b50d3d9a1759df0f8bb605cbafd3f33 Mon Sep 17 00:00:00 2001
From: Kacper Kornet <draenog@pld-linux.org>
Date: Wed, 27 Apr 2011 21:29:21 +0100
Subject: [PATCH 1/1] Create hooks directory if it is not created by git init

If git hooks directory in template is empty, the .git/hooks is not
created by git init. In this case it has to be created by gitolite.
---
 src/gitolite.pm |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/src/gitolite.pm b/src/gitolite.pm
index c0581c8..b4602a2 100644
--- a/src/gitolite.pm
+++ b/src/gitolite.pm
@@ -135,6 +135,7 @@ sub log_it {
 sub ln_sf
 {
     my($srcdir, $glob, $dstdir) = @_;
+    mkdir $dstdir unless (-d $dstdir);
     for my $hook ( glob("$srcdir/$glob") ) {
         $hook =~ s/$srcdir\///;
         unlink                   "$dstdir/$hook";
-- 
1.7.5

